sdk/identity/azure-identity/azure/identity/_internal/shared_token_cache.py

```diff
@@ -89,13 +90,21 @@ def _filtered_accounts(accounts, username=None, tenant_id=None):
 
 
 class SharedTokenCacheBase(ABC):
-    def __init__(self, username=None, **kwargs):  # pylint:disable=unused-argument
-        # type: (Optional[str], **Any) -> None
+    def __init__(self, username=None, profile=None, **kwargs):
```

Comment:
I am actually thinking if we can provide a helper for cross-tenant `get_token` or simple support `credential.get_token(tenant)`, since the `credential` already has everything we need, including `_profile`, `_username`, `environment`, etc.

```py
credential, msal_profile = InteractiveBrowserCredential.authenticate(
    client_id=_CLIENT_ID,
    silent_auth_only=True,
    scope='https://management.azure.com/.default'
)

    specific_tenant_credential = SharedTokenCacheCredential(username=credential._profile.username,
                                                            authority=credential._profile.environment,
                                                            tenant_id=tenant_id)
```

Reply:
You're correct the credential has enough information to authenticate in arbitrary tenants. However, credential and `msal` application instances are designed to authenticate in a single tenant. Revising that design for credentials requires careful thought and a (likely clumsy) workaround for `msal`.

I expect we'll keep the current design of a credential instance representing a single identity in a single tenant. For now, here's a way to create a credential instance per tenant:
```py
credential, profile = InteractiveBrowserCredential.authenticate(client_id=_CLIENT_ID)

for other_tenant in tenants:
    new_profile = AuthProfile(
        environment=profile.environment,
        home_account_id=profile.home_account_id,
        tenant_id=other_tenant,
        username=profile.username
    )
    new_credential = InteractiveBrowserCredential(
        client_id=_CLIENT_ID,
        profile=new_profile,
        silent_auth_only=True
    )
```
We can at least make this less awkward in a future revision.